### PR TITLE
Make rand optional

### PR DESCRIPTION
**Pull request by FreezyLemon**
After #270, making `rand` optional means that it's possible to use a very stripped-down version of the crate:

```
$ cargo tree -e normal --no-default-features
statrs v0.17.1
├── approx v0.5.1
│   └── num-traits v0.2.19
└── num-traits v0.2.19
```

Like the `nalgebra` feature, `rand` is enabled by default. This added modularity might make #165 easier, too.

This requires a breaking change: Removing the `rand::distribution::Distribution<T>` supertrait from `statrs::distribution::Distribution<T>` and `::DiscreteDistribution<T>`. Semantically, it means that the `statrs` distributions are not required to be `rand` distributions and are not logical extensions of them. Practically, it just means that new distributions don't necessarily have to implement the `rand` Distribution trait (all current distributions still do implement it with the feature enabled).

I would argue that the distribution traits still make sense without rand/RNG. This also allows using most of the `statistics` module without needing to compile `rand` and all that entails.

### CODE REVIEW

File: src/distribution/laplace.rs
[the remark this answers is not in the document]
Comment (FreezyLemon) [reply src/distribution/laplace.rs:1]: Looks like the context is #151. It's probably fine to just keep the test as-is, even if it's not a very involved test suite.

**Comment by YeungOnion** [comment]
Thanks again for what you've put in with the last 24 hours. It looks good, just gotta handle the merge.

**Comment by FreezyLemon** [comment]
This should be good to go. I resolved the merge conflicts manually, and the CI failure is just codecov

**Review by YeungOnion — APPROVED** [review]

**Comment by FreezyLemon** [comment]
Needed to rebase again after the other PRs were merged.

**Comment by YeungOnion** [comment]
coverage CI looks stuck. I'm messing around with it on poorly named pr #279

**Comment by YeungOnion** [comment]
Now, I will merge the correct thing. Whoops.

**Pull request merged**